service/history/replicatorQueueProcessor.go

```diff
@@ -99,6 +109,8 @@ func (p *replicatorQueueProcessorImpl) Process(qTask queueTaskInfo) error {
 
 	if err != nil {
 		p.metricsClient.IncCounter(scope, metrics.TaskRequests)
+	} else {
+		p.queueAckMgr.completeTask(task.TaskID)
```

Comment:
Why move this out of base processor?

Reply:
i believe there should be some logical separation of base processor and actual processor.
the base processor should only deal with retry, when seeing a error, and the actual task processor (the actual processing logic) should deal with errors such as entity not exists.